evalml/pipelines/time_series_classifcation_pipelines.py

```diff
@@ -0,0 +1,204 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.classification_pipeline import ClassificationPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    _get_rows_without_nans,
+    drop_rows_with_nans,
+    pad_with_nans
+)
+
+
+class TimeSeriesClassificationPipeline(ClassificationPipeline):
+    """Pipeline base class for time series classifcation problems."""
+
+    def __init__(self, parameters, random_state=0):
+        """Machine learning pipeline for time series regression problems made out of transformers and a classifier.
+
+        Required Class Variables:
+            component_graph (list): List of components in order. Accepts strings or ComponentBase subclasses in the list
+
+        Arguments:
+            parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
+                 An empty dictionary {} implies using all default values for component parameters. Pipeline-level
+                 parameters such as gap and max_delay must be specified with the "pipeline" key. For example:
+                 Pipeline(parameters={"pipeline": {"max_delay": 4, "gap": 2}}).
+            random_state (int, np.random.RandomState): The random seed/state. Defaults to 0.
+        """
+        if "pipeline" not in parameters:
+            raise ValueError("gap and max_delay parameters cannot be omitted from the parameters dict. "
+                             "Please specify them as a dictionary with the key 'pipeline'.")
+        pipeline_params = parameters["pipeline"]
+        self.gap = pipeline_params['gap']
+        self.max_delay = pipeline_params['max_delay']
+        super().__init__(parameters, random_state)
+
+    @staticmethod
+    def _convert_to_woodwork(X, y):
+        if X is None:
+            X = pd.DataFrame()
+        X = _convert_to_woodwork_structure(X)
+        y = _convert_to_woodwork_structure(y)
+        return X, y
+
+    def fit(self, X, y):
+        """Fit a time series regression pipeline.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): The input training data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray): The target training targets of length [n_samples]
+
+        Returns:
+            self
+        """
+        X, y = self._convert_to_woodwork(X, y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+        self._encoder.fit(y)
+        y = self._encode_targets(y)
```

Comment:
Throughout, we need to encode targets before computing features because we allow the user to pass in string-valued targets. If there is a `DelayedFeatureTransformer` in the pipeline, these string values will get added to the feature matrix and if there isn't a `OneHotEncoder`, the estimator will crash. 

Long story short, this is a repeat of #1477 . I think it would be better if we pass this complexity to the `DelayedFeatureTransformer` and have it encode the target before delaying but we should not let that block this PR. Happy to file something for updating the `DelayedFeatureTransformer`!

Reply:
Agree with you here! I like the idea of filing an issue and updating the `DelayedFeatureTransformer` as well as removing  encoding in the pipelines.

Reply:
When the dust settles and we build more features, I want to see if there is a way to refactor these classes to avoid duplication between classification and time series classification pipelines. 

It's not apparent right now because the "time-series-specific" code gets sprinkled in between the "non-time-series" code so it's not easy to just isolate the time series code into a helper function/mix-in yet. But maybe a new design will be clearer when this one becomes more stable and we test it!